src/server/routes/auth/keypair/import.ts

```diff
@@ -0,0 +1,83 @@
+import { Keypairs, Prisma } from "@prisma/client";
+import { Static, Type } from "@sinclair/typebox";
+import { FastifyInstance } from "fastify";
+import { StatusCodes } from "http-status-codes";
+import { insertKeypair } from "../../../../db/keypair/insert";
+import { createCustomError } from "../../../middleware/error";
+import {
+  KeypairAlgorithmSchema,
+  KeypairSchema,
+} from "../../../schemas/keypairs";
+import { standardResponseSchema } from "../../../schemas/sharedApiSchemas";
+
+const BodySchema = Type.Object({
+  publicKey: Type.String({
+    description:
+      "The public key of your keypair beginning with '-----BEGIN PUBLIC KEY-----'.",
+  }),
+  algorithm: KeypairAlgorithmSchema,
+});
+
+const ReplySchema = Type.Object({
+  result: Type.Object({
+    keypair: KeypairSchema,
+  }),
+});
+
+export async function importKeypair(fastify: FastifyInstance) {
+  fastify.route<{
+    Body: Static<typeof BodySchema>;
+    Reply: Static<typeof ReplySchema>;
+  }>({
+    method: "POST",
+    url: "/auth/keypair/import",
+    schema: {
+      summary: "Import public key",
+      description: "Import the public key for a keypair",
+      tags: ["Keypair"],
+      operationId: "import",
+      body: BodySchema,
+      response: {
+        ...standardResponseSchema,
+        [StatusCodes.OK]: ReplySchema,
+      },
+    },
+    handler: async (req, res) => {
+      const { publicKey, algorithm } = req.body;
```

Comment:
do we need to validate algorithm or we support all algorithm?

Reply:
The Typebox input schema enforces one of:
 "RS256",
  "RS384",
  "RS512",
  "ES256",
  "ES384",
  "ES512",
  "PS256",
  "PS384",
  "PS512",